ml_frameworks_bot/keras/retriever.py

```diff
@@ -97,25 +97,40 @@ def load_documents(
             repository_name,
             personal_access_token,
         )
+        # Build the sources if not already built
         source_directory = os.path.join(self.repository_local_path, "sources")
         if not os.path.exists(source_directory):
             build_keras_io_sources(repository_local_path=self.repository_local_path)
+
+        # Determine files for indexing
         input_files = []
         for directory in included_directories:
             input_files += get_all_file_paths(
                 os.path.join(self.repository_local_path, directory),
                 included_file_extensions=[".md"],
             )
+
+        # Load documents as TextNodes or Documents
         document_nodes = []
         for file_path in track(input_files, description="Loading documents:"):
+            # Exclude files with specific postfixes
             exclude_file = False
             for exclusion in exclude_file_postfixes:
                 if file_path.endswith(exclusion):
                     exclude_file = True
                     break
+
+            # Load the file content
             if not exclude_file:
                 with open(file_path, "r") as file:
                     text = file.read()
+
+                    # If file_path contains "ops/" and contains "----" then split based on "----"
+                    if file_path.contains("ops/"):
```

Comment:
A better paradigm might be pass this as a parameter in `KerasDocumentationRetreiver.load_documents` (and of course in `KerasDocumentationRetreiver.index_documents`. The paramater could pass a list of file patterns which would use this particular chunking strategy.